cfme/generic_objects/definition/ui.py

```diff
@@ -0,0 +1,477 @@
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == 'Add a new Custom Button' and self.in_generic_object_definition
+
+
+class GenericObjectButtonGroupAddView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    display = BootstrapSwitch(name='display')
+    description = Input(name='description')
+    image = VersionPick({
+        Version.lowest(): BootstrapSelect('button_image'),
+        '5.9': FonticonPicker('button_icon')})
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'GenericObjectButtonGroupAddView' and
+            self.in_generic_object_definition
+        )
+
+    def after_fill(self, was_change):
+        # we need to click somewhere out side the form to get add button active,
+        # after icon is filled
+        if was_change:
+            self.browser.element('//body').click()
+
+
+class GenericObjectButtonGroupDetailsView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+    configuration = Dropdown(text='Configuration')
+    basic_information = SummaryForm('Basic Information')
+    accordion = View.nested(AccordionForm)
+    table = Table('//h3[contains(text(), "Buttons")]/following-sibling::table')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.basic_information.is_displayed and
+            self.in_generic_object_definition and
+            'Custom Button Set' in self.title.text
+        )
+
+
+@MiqImplementationContext.external_for(GenericObjectDefinitionCollection.create, ViaUI)
+def create(self, name, description, attributes=None, associations=None, methods=None,
+           custom_image_file_path=None, cancel=False):
+    """
+    Create new generic object definition
+    Args:
+        name: generic object definition name
+        description: generic object definition description
+        attributes:  generic object definition attributes ex. {'address': 'string'}
+        associations:  generic object definition associations ex. {'services': 'Service'}
+        methods: generic object definition methods ex. ['method1', 'method2']
+        custom_image_file_path: generic object definition custom image file path
+        cancel: by default will not be canceled, pass True to make successful cancel
+
+    Returns:
+        GenericObjectDefinition entity
+
+    """
+    view = navigate_to(self, 'Add')
+    view.fill({
+        'name': name,
+        'description': description,
+        'associations': associations,
+        'attributes': attributes,
+        'methods': methods,
+        'custom_image_file': {'file': custom_image_file_path}
+    })
+    if cancel:
+        view.cancel.click()
+    else:
+        view.add.click()
+        view.flash.assert_no_error()
```

Comment:
Move this out of the if block, `assert_no_error` whether the form was cancelled or saved.